src/iso_alloc.c

```diff
@@ -2120,6 +2120,19 @@ INTERNAL_HIDDEN INLINE int _iso_getcpu(void) {
     const unsigned int cpunodesegment = 15 * 8 + 3;
     __asm__ volatile("lsl %1, %0" : "=r"(a) : "r"(cpunodesegment));
     return (int)(a & 0xfff);
+#elif defined(__aarch64__)
+#if defined(__APPLE__)
+    /* unlike other oses, the tpidr_el0 register on macOs is unused
+     * data stored for the current thread is instead fetchable
+     * from "tpidrro_el0".
+     */
+    uintptr_t a;
+    __asm__ volatile("mrs %x0, tpidrro_el0" : "=r"(a) :: "memory");
+    return (int)((a & 0x8) - 1);
+#else
+    /* TODO most likely different register/making on other platforms */
```

Comment:
For consistency with the rest of the code Id like to not use `defined()` or `ifdef` and instead use `#if` preprocessor which will only check for non-zero values.

If I am reading this right it will return -1 for Android on aarch64? I think we can can use SCHED_GETCPU on Android too. I can add that to the Android.mk file later if necessary though.